examples/nodejs/src/experimental-features/applicable-inputs.ts

```diff
@@ -0,0 +1,512 @@
+  }
+};
+
+const tokenKey = (token: Token) => {
+  return `${token.currency_symbol}-${token.token_name}`;
+};
+const accumulatorFromDeposit = (
+  env: Environment,
+  state: MarloweState,
+  action: CanDeposit
+) => {
+  const { party, into_account, of_token, deposits } = action.deposit;
+  const value = evalValue(env, state, deposits);
+
+  const depositKey = `${partyKey(party)}-${partyKey(into_account)}-${tokenKey(
+    of_token
+  )}-${value}`;
+  return {
+    deposits: { [depositKey]: action },
+    choices: {},
+    notifies: undefined,
+  };
+};
+
+const accumulatorFromChoice = (action: CanChoose) => {
+  const { for_choice } = action.choice;
+  const choiceKey = `${partyKey(for_choice.choice_owner)}-${
+    for_choice.choice_name
+  }`;
+  return {
+    deposits: {},
+    choices: { [choiceKey]: action },
+    notifies: undefined,
+  };
+};
+
+const accumulatorFromNotify = (action: CanNotify) => {
+  return {
+    deposits: {},
+    choices: {},
+    notifies: action,
+  };
+};
+
+function mergeBounds(bounds: Bound[]): Bound[] {
+  const mergedBounds: Bound[] = [];
+
+  const sortedBounds = [...bounds].sort((a, b) =>
+    a.from > b.from ? 1 : a.from < b.from ? -1 : 0
+  );
+
+  let currentBound: Bound | null = null;
+
+  for (const bound of sortedBounds) {
+    if (currentBound === null) {
+      currentBound = { ...bound };
+    } else {
+      if (bound.from <= currentBound.to) {
+        currentBound.to = Big.max(currentBound.to, bound.to);
+      } else {
+        mergedBounds.push(currentBound);
+        currentBound = { ...bound };
+      }
+    }
+  }
+
+  if (currentBound !== null) {
+    mergedBounds.push(currentBound);
+  }
+
+  return mergedBounds;
+}
+
+const flattenChoices = {
+  concat: (fst: CanChoose, snd: CanChoose): CanChoose => {
+    const mergedBounds = mergeBounds(
+      fst.choice.choose_between.concat(snd.choice.choose_between)
+    );
+    return {
+      type: "Choice",
+      policyId: fst.policyId,
+      choice: {
+        for_choice: fst.choice.for_choice,
+        choose_between: mergedBounds,
+      },
+      applyAction: (chosenNum: ChosenNum) => {
+        if (inBounds(chosenNum, fst.choice.choose_between)) {
+          return fst.applyAction(chosenNum);
+        } else {
+          return snd.applyAction(chosenNum);
+        }
+      },
+    };
+  },
+};
+
+const mergeApplicableActionAccumulator: Monoid<ApplicableActionAccumulator> = {
+  empty: {
+    deposits: {},
+    choices: {},
+    notifies: undefined,
+  },
+  concat: (fst, snd) => {
+    return {
+      deposits: R.union(flattenDeposits)(fst.deposits)(snd.deposits),
+      choices: R.union(flattenChoices)(fst.choices)(snd.choices),
+      notifies: fst.notifies ?? snd.notifies,
+    };
+  },
+};
+
+async function getApplicableActionFromCase(
+  restClient: RestClient,
+  env: Environment,
+  currentContract: Contract,
+  state: MarloweState,
+  previousPayments: Payment[],
+  previousWarnings: TransactionWarning[],
+  cse: Case,
```

Comment:
I bugged on `cse`... could it be something more `aCase` ? 
and  `cseContinuation` -> `aCaseContinuation`